providers/core.tcpsocket.js

```diff
@@ -296,15 +297,29 @@ Socket_chrome.prototype.dispatchDisconnect = function (code) {
 };
 
 /**
+ * Generates an index key for {@Socket_chrome.active}.
+ * The key is an integer equal to the socket ID for client sockets or, for
+ * server sockets, to the negation of the socket ID.
+ * @method keyForActive
+ * @private
+ * @param {Object} namespace Socket namespace, indicating client or server socket.
+ * @param {number} id The socketId, as provided by chrome.sockets.
+ * @static
+ * @see Socket_chrome.active
+ */
+Socket_chrome.keyForActive = function(namespace, id) {
+  return id * (namespace === chrome.sockets.tcp ? 1 : -1);
```

Comment:
Can you have the key be prefixed by namespace, rather than knowing that negative ones are servers? e.g. keys like
'server.1'  and 'client.2'